pkg/ccl/multitenantccl/tenantcostclient/token_bucket_test.go

```diff
@@ -116,6 +116,39 @@ func TestTokenBucket(t *testing.T) {
 	} else if exp := time.Nanosecond; tryAgainAfter != exp {
 		t.Fatalf("tryAgainAfter: expected %s, got %s", exp, tryAgainAfter)
 	}
+
+	// Set limit.
+	args = tokenBucketReconfigureArgs{
+		NewRate:   10,
+		NewTokens: 100,
+		MaxTokens: 50,
+	}
+	check("-30.00 RU filling @ +Inf RU/s")
+	tb.Reconfigure(ts.Now(), args)
+	check("50.00 RU filling @ 10.00 RU/s")
+	ts.Advance(time.Second)
+	check("50.00 RU filling @ 10.00 RU/s")
+	ts.Advance(time.Second)
+	tb.RemoveTokens(ts.Now(), 15)
+	check("35.00 RU filling @ 10.00 RU/s")
+	ts.Advance(2 * time.Second)
+	fulfill(10)
+	check("40.00 RU filling @ 10.00 RU/s")
+
+	// Fulfill amount > limit.
+	if ok, tryAgainAfter := tb.TryToFulfill(ts.Now(), 60); ok {
+		t.Fatalf("fulfilled incorrectly")
+	} else if exp := 2 * time.Second; tryAgainAfter.Round(time.Millisecond) != exp {
+		t.Fatalf("tryAgainAfter: expected %s, got %s", exp, tryAgainAfter)
+	}
```

Comment:
Interesting, it looks like if you have a request over the limit, you end up paying a bit of a penalty as you are asked to wait an amount of time that assumes no limit and then end up in debt for the request.  As you mention in your other comment, we don't expect it to happen so it seems fine to me.